### DISCUSSION TITLE
plugins execution sequence?

### DISCUSSION THREAD

**Discussion by redtide** [mkdocs/mkdocs#3276] [Q&A] [answered]
Excuse my ignorance, I would like to know how MKDocs performs the sequence of plugins (parallel or sequential).
That arose after the need of compile scss and minify both css and js assets, which it seems to be possible with 2 different plugins (mkdocs-extra-sass-plugin and mkdocs-minify-plugin), so the former could compile my style.scss to css, and the latter minify the result together with the scripts.

**Comment by pawamoy** [comment, marked as answer]
MkDocs respects the order of `plugins` in mkdocs.yml. However it also depends on the events that plugins handle or not: have a look at the diagram of events: https://www.mkdocs.org/dev-guide/plugins/#events.

Very, *very* rougly:

```python
for event in events:
    for plugin in plugins:
        plugin.handle(event)
```

(pseudocode)

**Reply by redtide** [reply]
Thank you! So this sounds like good news, I'll try the 2 above then and hope that together with a script to download some additional stuff, called from poetry, I can drop the need to build the assets using node and yarn.

**Comment by oprypin** [comment]
This is the documentation: https://www.mkdocs.org/dev-guide/plugins/#event-priorities

**Reply by redtide** [reply]
Thank you! I haven't noticed that part when searching.
